### DISCUSSION TITLE
Цвет команды при создании

### DISCUSSION THREAD

**Discussion by forwork10** [Sokol-Games-GmbH/PixelCombats.PublicApi#111] [Q&A] [answered]
Цвет команды получается противоположный, тоесть на него применяется формула color={255-r,255-g,255-b};
Из-за этого многие не понимают почему они ставят один цвет, а получается другой.
Также в структуре color есть параметр прозрачности, хотелось бы видеть его в работе

**Comment by Kukushechka7-ezx** [comment]
Нет... Он делится на 255, r/255

**Comment by forwork10** [comment]
вообще-то нет... ты проверь сам)

**Comment by Kukushechka7-ezx** [comment]
Проверил, и чё? Разделил цвет на 255 и вышел тот который в инете был,лол

**Comment by forwork10** [comment]
не проверил ты..

**Comment by Kukushechka7-ezx** [comment]
@kkohno ну скажите уже)

**Comment by Kukushechka7-ezx** [comment]
![Screenshot_2023-09-14-17-43-44-843_com android chrome](https://github.com/Sokol-Games-GmbH/PixelCombats.PublicApi/assets/142585063/467bd02c-8496-41d5-8a94-a32ff06cc20e)
Ну как с таким человеком спорить?

**Comment by kkohno** [comment, marked as answer]
[Вот файл структуры а АПИ, описывающий цвет](https://github.com/Sokol-Games-GmbH/PixelCombats.PublicApi/blob/main/Api/Basic/Structures/Math/Color.cs). Как видно из кода:
```csharp
	[Serializable]
	public struct Color
	{
		[SerializeMember(1)]
		public float r;
		[SerializeMember(2)]
		public float g;
		[SerializeMember(3)]
		public float b;
		[SerializeMember(4)]
		public float a;
```
Тут float это вещественные числа, тоесть не целые. Обычно при таком раскладе макс интенсивность составляющей цвета это 1 а минимум 0, то-есть задаем каждую в диапазоне [0-1].
Нам следовало бы написать комментарии к этому делу в коде, вставим.
По альфе - она есть. но запланирована для случаев, когда нужна прозрачность. Однако прозрачность на мобильних устройствах это дорого, поэтому редко юзается.

**Comment by forwork10** [comment]
> ![Screenshot_2023-09-14-17-43-44-843_com android chrome](https://github.com/Sokol-Games-GmbH/PixelCombats.PublicApi/assets/142585063/467bd02c-8496-41d5-8a94-a32ff06cc20e)
> Ну как с таким человеком спорить? 

ну уж прости, я привык что ты всегда не прав)
